traffic_portal/app/src/common/modules/form/user/FormUserController.js

```diff
@@ -37,6 +37,16 @@ var FormUserController = function(user, $scope, $location, formUtils, stringUtil
             });
     };
 
+    $scope.changeLogs = [];
+
+    $scope.getChangeLogs = function(userName) {
+        changeLogService.getChangeLogs({ username: userName })
+            .then(function(response) {
+                $scope.changeLogs = response;
+                locationUtils.navigateToPath('/change-logs?username='+userName);
```

Comment:
`userName` should be sanitized with [`encodeURIComponent`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/encodeURIComponent).

Reply:
ok but why?

Reply:
Because if the username contains <kbd>;</kbd>, <kbd>,</kbd>, <kbd>/</kbd>, <kbd>?</kbd>, <kbd>:</kbd>, <kbd>@</kbd>, <kbd>&</kbd>, <kbd>=</kbd>, <kbd>+</kbd>, <kbd>$</kbd>, or <kbd> </kbd> the URL will be parsed incorrectly - or if the username contains any UTF8 codepoints that begin or end with bytes that could be misinterpreted as those characters. Reserved characters and non-ascii code-points *must* be properly encoded in URLs, **especially** when they come from user input, which usernames do. It's a possible attack vector.

Reply:
Got it..since I changed the button to a link, we no longer need this function and its sanitization.